Document pins v2 (#384)
discord-userdoccers/discord-userdoccers: pages/resources/message.mdx

| -------- | ---------------- | --------------------------------- |
| messages | array[snowflake] | The message IDs to delete (2-100) |

<RouteHeader method="GET" url="/channels/{channel.id}/pins">

>   Get Pinned Messages

Reviewer: keep the old get route imo

| Field   | Type              | Description                                     |
| ------- | ----------------- | ----------------------------------------------- |
| before? | ISO8601 timestamp | Get messages pinned before this timestamp       |
| after?  | ISO8601 timestamp | Get messages pinned after this timestamp        |
| limit?  | integer           | Max number of pins to return (1-50, default 50) |

###### Response Body

| Field    | Type                                             | Description                                                                                       |
| -------- | ------------------------------------------------ | ------------------------------------------------------------------------------------------------- |
| items    | array[[message pin](#message-pin-object) object] | The message pins in the channel                                                                   |
| has_more | boolean                                          | Whether there are potentially additional message pins that could be returned on a subsequent call |

<RouteHeader method="PUT" url="/channels/{channel.id}/messages/pins/{message.id}" supportsAuditReason>

>   Add Message Pin

Reviewer: ```suggestion
  Pin Message
```

</RouteHeader>

Pins a message in a channel. Requires the `MANAGE_MESSAGES` permission if operating on a guild channel. Returns a 204 empty response on success. Fires a [Channel Pins Update](/topics/gateway-events#channel-pins-update) Gateway event.

<RouteHeader method="DELETE" url="/channels/{channel.id}/messages/pins/{message.id}" supportsAuditReason>

>   Remove Message Pin

Reviewer: ```suggestion
  Unpin Message
```

</Alert>

###### Query String Params

| Field   | Type              | Description                                     |
| ------- | ----------------- | ----------------------------------------------- |
| before? | ISO8601 timestamp | Get messages pinned before this timestamp       |
| limit?  | integer           | Max number of pins to return (1-50, default 50) |

###### Response Body

| Field    | Type                                             | Description                                                                                       |
| -------- | ------------------------------------------------ | ------------------------------------------------------------------------------------------------- |
| items    | array[[message pin](#message-pin-object) object] | The message pins in the channel                                                                   |
| has_more | boolean                                          | Whether there are potentially additional message pins that could be returned on a subsequent call |

<RouteHeader method="GET" url="/channels/{channel.id}/pins" deprecated>
  Get Pinned Messages
</RouteHeader>

> Returns all pinned messages in the channel as an array of [message](#message-object) objects without the `reactions` key.

Reviewer: ```suggestion
Returns up to 50 pinned messages in the channel as an array of [message](#message-object) objects without the `reactions` key.
```

| -------- | ---------------- | --------------------------------- |
| messages | array[snowflake] | The message IDs to delete (2-100) |

<RouteHeader method="GET" url="/channels/{channel.id}/messages/pins">
  Get Message Pins
</RouteHeader>

Returns all message pins in the channel. Requires the `VIEW_CHANNEL` permission.

<Alert type="note">

If the current user does not have `READ_MESSAGE_HISTORY` permission, an empty ``items`` array will be returned.

</Alert>

###### Query String Params

| Field   | Type              | Description                                     |
| ------- | ----------------- | ----------------------------------------------- |
| before? | ISO8601 timestamp | Get messages pinned before this timestamp       |

> | after?  | ISO8601 timestamp | Get messages pinned after this timestamp        |

Reviewer: ```suggestion
```
